react/components/ShippingSimulator/Wrapper.js

```diff
@@ -160,6 +160,7 @@ const ShippingSimulatorWithOrderForm = ({
 
   return (
     <BaseShippingSimulatorWrapper
+      key={seller}
```

Comment:
This solution is tricky and solves the problem, but in my opinion, it can be solved with a better approach without destroying and remount the whole component tree.

The problem is, when changing the location code, should change the table of shipping price? If yes, why not create an effect (can be useEffect) to handle this property change?

Reply:
Apparently the component already is remounted, but the `useRef` control the `useEffect` execution and prevents the `handleCalculateShipping()` from being called.

https://github.com/vtex-apps/store-components/blob/efebaee9e9e981849018d1f3d224f948ab7a521b/react/components/ShippingSimulator/Wrapper.js#L115-L127

My problem happens when the `sellerId` changes. Currently when the `sellerId` changes, the Shipping Simulator don't call the `handleCalculateShipping` with this new value. So, the Seller Table component to be continued showing the old values, you know?
It's happens because my new component [buybox-context](https://github.com/vtex-apps/buybox-context) can change the seller as "winner" of buybox. And the shipping simulator impact diretly on this, because if a seller has a better price with Price + Shipping Price, the seller can be changed based on this new values.

I simulated your suggestion to compare the solutions using the Profiller of React Developer Tools, and apparently the execution cost was higher.
I put this code on `Wrapper.js`, as you can see bellow:

```
  useEffect(() => {
    if (isMountedRef.current) {
      isMountedRef.current = false
    }
  }, [seller])
```

https://github.com/vtex-apps/store-components/blob/efebaee9e9e981849018d1f3d224f948ab7a521b/react/components/ShippingSimulator/Wrapper.js#L113-L114

Reply:
Here they are exported profile that I used to compare the performance render of `BaseShippingSimulatorWrapper` component

[BaseShippingSimulatorWrapper.zip](https://github.com/vtex-apps/store-components/files/7261040/BaseShippingSimulatorWrapper.zip)

Reply:
Hey @igorbrasileiro, can you help me? 😄